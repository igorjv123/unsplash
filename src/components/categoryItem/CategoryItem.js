import React from 'react';
import './style.sass';
import {shape, func, string, array, arrayOf} from 'prop-types';

export const CategoryItem = (props) => {

    const { id, title, description, preview_photos } = props.item;
    const handleClick = () => {
        props.onClick(id)
    }
    return(
      <div className={'category-item'} onClick={ handleClick }>
          <img className={'category-item__image'} src={preview_photos[0].urls.small} alt={description}/>
          <h2 className={'category-item__title'}>
              <span>{title}</span>
          </h2>
      </div>
    );
}

CategoryItem.propTypes = {
    item: shape({
        id: string.isRequired,
        title: string.isRequired,
        description: string.isRequired,
        preview_photos: arrayOf(shape({
            urls: shape({
                small: string.isRequired
            })
        })
        )
    }).isRequired,
    onClick: func.isRequired
}

CategoryItem.defaultProps = {
    item: {
        id: 'id',
        title: 'title',
        description: 'desc',
        preview_photos: [{urls: {
            small: 'www.default-link.com'
        }
        }]
    },
    onClick: () => {}
}

export default CategoryItem;