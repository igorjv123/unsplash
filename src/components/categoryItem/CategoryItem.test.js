import React from 'react';
import { CategoryItem } from "./CategoryItem";
import { render, mount } from 'enzyme';

describe('CategoryItem ', () => {
    const onClick = jest.fn();
    const categoryMock = {
        id: 'id',
        title: 'title',
        description: 'desc',
        preview_photos: [{urls: {
                small: 'www.default-link.com'
            }
        }]
    };

    beforeEach(function () {
        onClick.mockClear();
    });

   it('should render without crashing', () => {
      const wrapper = render(<CategoryItem onClick={onClick} item={categoryMock}/>);
      expect(wrapper).toMatchSnapshot();
   });

   it('should correctly handle onClick', () => {
       const wrapper = mount(<CategoryItem image={categoryMock} onClick={onClick}/>)

       wrapper.find('.category-item').simulate('click');

       expect(onClick).toHaveBeenCalled();
       expect(onClick).toHaveBeenCalledWith(categoryMock.id);
   });
});