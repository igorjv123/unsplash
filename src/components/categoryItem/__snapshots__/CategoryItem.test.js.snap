// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem  should render without crashing 1`] = `
<div
  class="category-item"
>
  <img
    alt="desc"
    class="category-item__image"
    src="www.default-link.com"
  />
  <h2
    class="category-item__title"
  >
    <span>
      title
    </span>
  </h2>
</div>
`;
